source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,17 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	vector<shared_ptr<Ship>>::iterator firstShipIt = ships.begin();
```

Comment:
You should really use auto, it's more readable and everyone knows what begin() returns

Reply:
Alright. I just thought as it says in the wiki, to try to avoid auto as good as possible :x